File: android/app/src/main/java/online/db1k/safering/android/data/remote/models/PrefixResponse.kt
package online.db1k.safering.android.data.remote.models

import com.google.gson.annotations.SerializedName

/**
 * Response from GET /v1/prefixes
 * Mirrors the iOS PrefixResponse.swift exactly.
 */
data class PrefixResponse(
    @SerializedName("prefixes") val prefixes: List<ScamPrefix>,
    @SerializedName("updated_at") val updatedAt: Long
)

data class ScamPrefix(
    @SerializedName("prefix") val prefix: String,
    @SerializedName("risk") val risk: Double,
    @SerializedName("count") val count: Int,
    @SerializedName("common_tags") val commonTags: List<String>
)

File: ios/SafeRing/Data/Remote/Models/PrefixResponse.swift
import Foundation

/// Response from GET /v1/prefixes
/// Contains known scammer phone number prefixes for offline prefix-based blocking.
struct PrefixResponse: Decodable {
    /// Array of known scam prefixes and their associated risk levels.
    let prefixes: [ScamPrefix]

    /// Unix timestamp indicating when this prefix data was compiled.
    let updatedAt: TimeInterval

    enum CodingKeys: String, CodingKey {
        case prefixes
        case updatedAt = "updated_at"
    }
}

/// A known scam phone number prefix pattern.
struct ScamPrefix: Decodable {
    /// The prefix string (e.g., "+1234", "123", "+1-800-")
    let prefix: String

    /// Risk score associated with numbers matching this prefix.
    let risk: Double

    /// Number of known scam numbers matching this prefix.
    let count: Int

    /// Common scam types associated with this prefix.
    let commonTags: [String]

    enum CodingKeys: String, CodingKey {
        case prefix, risk, count
        case commonTags = "common_tags"
    }
}
